src/edge/simulator.ts

```diff
@@ -226,10 +194,148 @@ export class Simulator {
         });
     }
 
+    private async getLastestStandaloneSimulatorInfo() {
+        if (!this.latestStandaloneSimulatorInfoJson) {
+            const infoString = await request.get(Simulator.latestReleaseInfoUrl, {
+                headers: {
+                    "User-Agent": "vscode-azure-iot-edge",
+                },
+            });
+            const infoJson = JSON.parse(infoString);
+            this.latestStandaloneSimulatorInfoJson = infoJson;
+            return infoJson;
+        } else {
+            return this.latestStandaloneSimulatorInfoJson;
+        }
+    }
+
+    private async getSimulatorType(): Promise<SimulatorType> {
+        if (await this.simulatorInstalled()) {
```

Comment:
seems simulatorInstalled  method should return whether standalone or pip, instead of re-check the platform here.

Reply:
updated